lib/ansible/modules/systemd.py

```diff
@@ -497,11 +497,21 @@ def main():
 
             # do we need to enable the service?
             enabled = False
-            (rc, out, err) = module.run_command("%s is-enabled '%s'" % (systemctl, unit))
+            (rc, out, err) = module.run_command("%s is-enabled '%s' -l" % (systemctl, unit))
 
             # check systemctl result or if it is a init script
             if rc == 0:
-                enabled = True
+                # check if service is indirect
+                if out.splitlines()[0] == 'indirect':
+                    # get the service that should be targeted for enable
+                    indirect_unit = out.split()[-1].split('/')[-1]
```

Comment:
Should we only check the target if it's included in stdout? If stdout is just `indirect\n\n`, the command below ends up as `... is-enabled 'indirect' -l`.

Reply:
I actually think reworking this like:

```python3
       # check systemctl result or if it is a init script
            if rc == 0:
                # check if service is indirect
                if out.splitlines()[0] == "indirect":
                    # if is-enabled -l has more than 1 line its already enabled
                    if len(out.splitlines()) > 1:
                        enabled = True
                else:
                    enabled = True
```

I think my idea of getting the indirect target is not needed...something naive like the above is better... if an indirect status outputs exactly "indirect" and is just one line than the service is disabled:

```
[root@fedora-s-1vcpu-1gb-nyc1-01 ~]# systemctl disable tftp
Removed /etc/systemd/system/sockets.target.wants/tftp.socket.
[root@fedora-s-1vcpu-1gb-nyc1-01 ~]# systemctl is-enabled tftp  -l
indirect
```

If its output > 2 lines its enabled as it has the target unit:

```
[root@fedora-s-1vcpu-1gb-nyc1-01 ~]# systemctl enable tftp
Created symlink /etc/systemd/system/sockets.target.wants/tftp.socket → /usr/lib/systemd/system/tftp.socket.
[root@fedora-s-1vcpu-1gb-nyc1-01 ~]# systemctl is-enabled tftp  -l
indirect
  /etc/systemd/system/sockets.target.wants/tftp.socket
[root@fedora-s-1vcpu-1gb-nyc1-01 ~]# 

```

What do you think ?

Reply:
That seems right -  no need to make the second call to run_command if the target is in the output.
IIRC, when I tested before briefly, I thought there were two newlines when not enabled. `out` was `"indirect\n\n"`, so out.splitlines() == ['indirect', '']. If that's the case,  the code could check if the second line is truthy.

Reply:
OK done.. from all my tests when its disabled it will just be:

```
indirect
```

and when enabled:

```
indirect
 /path/to/some/service/here
```